### ISSUE TITLE
fixturenet-eth startup failure

### ISSUE THREAD

**Issue by dboreham** [cerc-io/stack-orchestrator#333]
Service `fixturenet-eth-geth-2` failed to start, I think because it depends on `fixturenet-eth-bootnode-geth` which was not yet up at the time it tried to contact it.

```
$ docker logs laconic-2823003388f3f8eee1aa740ece27664c-fixturenet-eth-geth-2-1

0xe6CE22afe802CAf5fF7d3845cec8c736ecc8d61F
INFO [04-16|17:32:06.859] Maximum peer count                       ETH=50 LES=0 total=50
INFO [04-16|17:32:06.859] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [04-16|17:32:06.861] Set global gas cap                       cap=50,000,000
Address: {e6ce22afe802caf5ff7d3845cec8c736ecc8d61f}

0xe22AD83A0dE117bA0d03d5E94Eb4E0d80a69C62a
INFO [04-16|17:32:08.995] Maximum peer count                       ETH=50 LES=0 total=50
INFO [04-16|17:32:08.996] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [04-16|17:32:08.996] Set global gas cap                       cap=50,000,000
Address: {e22ad83a0de117ba0d03d5e94eb4e0d80a69c62a}

0xf1ac8Dd1f6D6F5c0dA99097c57ebF50CD99Ce293
INFO [04-16|17:32:10.638] Maximum peer count                       ETH=50 LES=0 total=50
INFO [04-16|17:32:10.638] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [04-16|17:32:10.639] Set global gas cap                       cap=50,000,000
Address: {f1ac8dd1f6d6f5c0da99097c57ebf50cd99ce293}

0x9d2edB2B30Bce41375179571944A3f92636ce1Cd
INFO [04-16|17:32:12.235] Maximum peer count                       ETH=50 LES=0 total=50
INFO [04-16|17:32:12.235] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [04-16|17:32:12.236] Set global gas cap                       cap=50,000,000
Address: {9d2edb2b30bce41375179571944a3f92636ce1cd}

0x5D81E609c15E292Bb8255bd9b1B2494DC0386062
INFO [04-16|17:32:13.894] Maximum peer count                       ETH=50 LES=0 total=50
INFO [04-16|17:32:13.894] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [04-16|17:32:13.895] Set global gas cap                       cap=50,000,000
Address: {5d81e609c15e292bb8255bd9b1b2494dc0386062}

0x5929AD4A1d6B899065AcF2a66d5EB086a2863bEe
INFO [04-16|17:32:15.557] Maximum peer count                       ETH=50 LES=0 total=50
INFO [04-16|17:32:15.557] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [04-16|17:32:15.558] Set global gas cap                       cap=50,000,000
Address: {5929ad4a1d6b899065acf2a66d5eb086a2863bee}
INFO [04-16|17:32:17.181] Enabling metrics collection
INFO [04-16|17:32:17.181] Enabling stand-alone metrics HTTP endpoint address=0.0.0.0:6060
INFO [04-16|17:32:17.181] Starting metrics server                  addr=http://0.0.0.0:6060/debug/metrics
CRIT [04-16|17:32:19.830] Bootstrap URL invalid                    enode=enode://af22c29c316ad069cf48a09a4ad5cf04a251b411e45098888d114c6dd7f489a13786620d5953738762afa13711d4ffb3b19aa5de772d8af72f851f7e9c5b164a@fixturenet-eth-bootnode-geth:30303 err="lookup fixturenet-eth-bootnode-geth on 127.0.0.11:53: no such host"
```

**Comment by dboreham** [comment]
This seems to be worse than just container startup ordering. For some reason the whole cluster is broken from a DNS perspective.

**Comment by dboreham** [comment]
Problem seems to be that the bootnode-geth container exits after startup:

```
$ docker logs laconic-2823003388f3f8eee1aa740ece27664c-fixturenet-eth-bootnode-geth-1
INFO [04-16|18:05:53.967] Maximum peer count                       ETH=50 LES=0 total=50
INFO [04-16|18:05:53.969] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [04-16|18:05:53.974] Set global gas cap                       cap=50,000,000
INFO [04-16|18:05:53.976] Allocated trie memory caches             clean=154.00MiB dirty=256.00MiB
INFO [04-16|18:05:53.976] Allocated cache and file handles         database=/root/.ethereum/geth/chaindata cache=512.00MiB handles=524,288
INFO [04-16|18:05:54.058] Opened ancient database                  database=/root/.ethereum/geth/chaindata/ancient/chain readonly=false
INFO [04-16|18:05:54.059]
INFO [04-16|18:05:54.059] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [04-16|18:05:54.059] Chain ID:  1212 (unknown)
INFO [04-16|18:05:54.059] Consensus: unknown
INFO [04-16|18:05:54.059]
INFO [04-16|18:05:54.059] Pre-Merge hard forks:
INFO [04-16|18:05:54.059]  - Homestead:                   1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/homestead.md)
INFO [04-16|18:05:54.059]  - Tangerine Whistle (EIP 150): 1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/tangerine-whistle.md)
INFO [04-16|18:05:54.059]  - Spurious Dragon/1 (EIP 155): 1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [04-16|18:05:54.059]  - Spurious Dragon/2 (EIP 158): 1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [04-16|18:05:54.059]  - Byzantium:                   1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/byzantium.md)
INFO [04-16|18:05:54.059]  - Constantinople:              1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/constantinople.md)
INFO [04-16|18:05:54.059]  - Petersburg:                  1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/petersburg.md)
INFO [04-16|18:05:54.059]  - Istanbul:                    1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/istanbul.md)
INFO [04-16|18:05:54.059]  - Berlin:                      1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/berlin.md)
INFO [04-16|18:05:54.059]  - London:                      1        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/london.md)
INFO [04-16|18:05:54.059]
INFO [04-16|18:05:54.059] Merge configured:
INFO [04-16|18:05:54.059]  - Hard-fork specification:    https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/paris.md
INFO [04-16|18:05:54.059]  - Network known to be merged: false
INFO [04-16|18:05:54.059]  - Total terminal difficulty:  1000
INFO [04-16|18:05:54.059]  - Merge netsplit block:       <nil>
INFO [04-16|18:05:54.059] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [04-16|18:05:54.059]
INFO [04-16|18:05:54.060] Disk storage enabled for ethash caches   dir=/root/.ethereum/geth/ethash count=3
INFO [04-16|18:05:54.060] Disk storage enabled for ethash DAGs     dir=/root/.ethash               count=2
INFO [04-16|18:05:54.060] Initialising Ethereum protocol           network=1212 dbversion=<nil>
INFO [04-16|18:05:54.061] Loaded most recent local header          number=0 hash=8a3c7c..042614 td=1 age=54y3w2d
INFO [04-16|18:05:54.061] Loaded most recent local full block      number=0 hash=8a3c7c..042614 td=1 age=54y3w2d
INFO [04-16|18:05:54.061] Loaded most recent local fast block      number=0 hash=8a3c7c..042614 td=1 age=54y3w2d
WARN [04-16|18:05:54.061] Failed to load snapshot, regenerating    err="missing or corrupted snapshot"
INFO [04-16|18:05:54.061] Rebuilding state snapshot
INFO [04-16|18:05:54.061] Resuming state snapshot generation       root=151cf8..28a98b accounts=0 slots=0 storage=0.00B dangling=0 elapsed="267.901µs"
INFO [04-16|18:05:54.062] Regenerated local transaction journal    transactions=0 accounts=0
WARN [04-16|18:05:54.062] Chain pre-merge, sync via PoW (ensure beacon client is ready)
INFO [04-16|18:05:54.062] Gasprice oracle is ignoring threshold set threshold=2
WARN [04-16|18:05:54.062] Error reading unclean shutdown markers   error="leveldb: not found"
WARN [04-16|18:05:54.062] Engine API enabled                       protocol=eth
INFO [04-16|18:05:54.063] Starting peer-to-peer node               instance=Geth/v1.10.23-statediff-4.2.0-alpha-bb20bec0-20221010/linux-amd64/go1.18.5
INFO [04-16|18:05:54.068] Generated state snapshot                 accounts=263 slots=31 storage=12.85KiB dangling=0 elapsed=6.792ms
INFO [04-16|18:05:54.122] New local node record                    seq=1,681,668,354,080 id=f19e412334ea6bbc ip=127.0.0.1 udp=0 tcp=30303
INFO [04-16|18:05:54.123] Started P2P networking                   self="enode://af22c29c316ad069cf48a09a4ad5cf04a251b411e45098888d114c6dd7f489a13786620d5953738762afa13711d4ffb3b19aa5de772d8af72f851f7e9c5b164a@127.0.0.1:30303?discport=0"
INFO [04-16|18:05:54.123] Generated JWT secret                     path=/root/.ethereum/geth/jwtsecret
INFO [04-16|18:05:54.143] WebSocket enabled                        url=ws://127.0.0.1:8551
INFO [04-16|18:05:54.143] HTTP server started                      endpoint=127.0.0.1:8551 auth=true prefix= cors=localhost vhosts=localhost
ERROR[04-16|18:05:54.144] Low disk space. Gracefully shutting down Geth to prevent database corruption. available=0.00B
INFO [04-16|18:05:54.144] Got interrupt, shutting down...
WARN [04-16|18:05:54.145] HTTP server graceful shutdown timed out
INFO [04-16|18:05:54.145] HTTP server stopped                      endpoint=127.0.0.1:8551
INFO [04-16|18:05:54.145] Ethereum protocol stopped
INFO [04-16|18:05:54.146] Transaction pool stopped
INFO [04-16|18:05:54.146] Writing snapshot state to disk           root=151cf8..28a98b
INFO [04-16|18:05:54.146] Persisted trie from memory database      nodes=0 size=0.00B time="5.6µs" gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=0.00B
INFO [04-16|18:05:54.146] Writing clean trie cache to disk         path=/root/.ethereum/geth/triecache threads=16
INFO [04-16|18:05:54.149] Persisted the clean trie cache           path=/root/.ethereum/geth/triecache elapsed=3.387ms
INFO [04-16|18:05:54.149] Blockchain stopped
```

**Comment by dboreham** [comment]
Reason is this:

```
ERROR[04-16|18:05:54.144] Low disk space. Gracefully shutting down Geth to prevent database corruption. available=0.00B
```

**Comment by dboreham** [comment]
This was due to Docker being fubar on my machine.
